src/mdast-process-images.js

```diff
@@ -21,14 +21,37 @@ export class TooManyImagesError extends Error {
  * @param {object} tree
  * @param {MediaHandler} mediaHandler
  * @param {string} baseUrl
+ * @param {Array<string>} externalImageUrlPrefixes Array of url prefixes to detect external images
  */
-export async function processImages(log, tree, mediaHandler, baseUrl) {
+export async function processImages(
+  log,
+  tree,
+  mediaHandler,
+  baseUrl,
+  externalImageUrlPrefixes = [],
+) {
   if (!mediaHandler) {
     return;
   }
   // gather all image nodes
   const images = new Map();
+  // Convert externalImageUrlPrefixes to an array if not already
+  if (!Array.isArray(externalImageUrlPrefixes)) {
+    // eslint-disable-next-line no-param-reassign
+    externalImageUrlPrefixes = [externalImageUrlPrefixes];
+  }
+
   const register = (node) => {
+    // Check if this is an external image
+    const { url = '' } = node;
```

Comment:
you extract url here, but then don't use it later (line 55/56)

Reply:
yes, that was pre-existing but good to use `url` there as well.